File: app/src/test/java/ru/p3tr0vich/fuel/utils/UtilsStringTest.java
package ru.p3tr0vich.fuel.utils;

import android.support.annotation.NonNull;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class UtilsStringTest {

    @NonNull
    private String doEncodeDecode(String s) {
        return UtilsString.decodeLineBreaks(UtilsString.encodeLineBreaks(s));
    }

    private void checkAssert(String s) throws AssertionError {
        assertEquals(s, s, doEncodeDecode(s));
    }

    @Test
    public void testEncodeDecode() throws AssertionError {
        checkAssert("\n");

        checkAssert("\\n");

        checkAssert("xxx\nyyy");

        checkAssert("xxx\\nyyy");

        checkAssert("xxx\\n\nyyy");
    }
}

File: app/src/test/kotlin/ru/p3tr0vich/fuel/UtilsStringTest.kt
package ru.p3tr0vich.fuel

import org.junit.Assert.assertEquals
import org.junit.Test
import ru.p3tr0vich.fuel.utils.UtilsString

class UtilsStringTest {

    private fun doEncodeDecode(s: String): String {
        return UtilsString.decodeLineBreaks(UtilsString.encodeLineBreaks(s))
    }

    @Throws(AssertionError::class)
    private fun checkAssert(s: String) {
        assertEquals(s, s, doEncodeDecode(s))
    }

    @Test
    @Throws(AssertionError::class)
    fun testEncodeDecode() {
        checkAssert("\n")

        checkAssert("\\n")

        checkAssert("xxx\nyyy")

        checkAssert("xxx\\nyyy")

        checkAssert("xxx\\n\nyyy")
    }
}